Cubical/Structures/Semigroup.agda

```diff
@@ -3,80 +3,168 @@ module Cubical.Structures.Semigroup where
+  SemigroupPath' = SIP semigroup-is-SNS
+
+  SemigroupIsoTemp : (M N : Semigroup {ℓ}) → Type ℓ
+  SemigroupIsoTemp M N = Semigroup→Semigroup' M ≃[ semigroup-iso ] Semigroup→Semigroup' N
+
+  SemigroupIsoTempPath : (M N : Semigroup {ℓ}) → Iso (SemigroupIso M N) (SemigroupIsoTemp M N)
+  fun (SemigroupIsoTempPath M N) record { equiv = x ; is-hom = y } = (x , y)
+  inv (SemigroupIsoTempPath M N) (x , y) = record { equiv = x ; is-hom = y }
+  rightInv (SemigroupIsoTempPath M N) _ = refl
+  leftInv (SemigroupIsoTempPath M N) _  = refl
+
+  SemigroupPath : (M N : Semigroup {ℓ}) → (SemigroupIso M N) ≃ (M ≡ N)
+  SemigroupPath M N =
+    SemigroupIso M N                                ≃⟨ isoToEquiv (SemigroupIsoTempPath M N) ⟩
+    SemigroupIsoTemp M N                            ≃⟨ SemigroupPath' (Semigroup→Semigroup' M) (Semigroup→Semigroup' N) ⟩
+    Semigroup→Semigroup' M ≡ Semigroup→Semigroup' N ≃⟨ isoToEquiv (invIso (congIso SemigroupIsoSemigroup')) ⟩
+    M ≡ N ■
+
+
+open semigroup-sip
+
+
+open Iso
+
+
+-- TODO: do we need any of the code below now that we have a Record?
+
+-- Operations for extracting components
 
-open semigroup-operation-syntax
+-- ⟨_⟩ : Semigroup' → Type ℓ
+-- ⟨ G , _ ⟩ = G
 
-semigroup-isSet : (G : Semigroup {ℓ}) → isSet ⟨ G ⟩
-semigroup-isSet (_ , _ , P , _) = P
+-- semigroup-operation : (G : Semigroup' {ℓ}) → ⟨ G ⟩ → ⟨ G ⟩ → ⟨ G ⟩
+-- semigroup-operation (_ , f , _) = f
 
-semigroup-assoc : (G : Semigroup {ℓ})
-                → (x y z : ⟨ G ⟩) → (x ·⟨ G ⟩ (y ·⟨ G ⟩ z)) ≡ ((x ·⟨ G ⟩ y) ·⟨ G ⟩ z)
-semigroup-assoc (_ , _ , _ , P) = P
+-- module semigroup-operation-syntax where
 
--- Semigroup equivalences
+--   semigroup-operation-syntax : (G : Semigroup' {ℓ}) → ⟨ G ⟩ → ⟨ G ⟩ → ⟨ G ⟩
+--   semigroup-operation-syntax = semigroup-operation
 
-semigroup-iso : StrIso semigroup-structure ℓ
-semigroup-iso = add-to-iso (binaryFunIso pointed-iso) semigroup-axioms
+--   infixl 20 semigroup-operation-syntax
+--   syntax semigroup-operation-syntax G x y = x ·⟨ G ⟩ y
 
-semigroup-axiom-isProp : (X : Type ℓ)
-                       → (s : raw-semigroup-structure X)
-                       → isProp (semigroup-axioms X s)
-semigroup-axiom-isProp X _·_ = isPropΣ isPropIsSet
-                               λ isSetX →  isPropΠ (λ x → isPropΠ (λ y → isPropΠ (λ z → isSetX _ _)))
+-- open semigroup-operation-syntax
 
-semigroup-is-SNS : SNS {ℓ} semigroup-structure semigroup-iso
-semigroup-is-SNS = add-axioms-SNS _ semigroup-axiom-isProp raw-semigroup-is-SNS
+-- semigroup-isSet : (G : Semigroup' {ℓ}) → isSet ⟨ G ⟩
+-- semigroup-isSet (_ , _ , P , _) = P
 
-SemigroupPath : (M N : Semigroup {ℓ}) → (M ≃[ semigroup-iso ] N) ≃ (M ≡ N)
-SemigroupPath = SIP semigroup-is-SNS
+-- semigroup-assoc : (G : Semigroup' {ℓ})
+--                 → (x y z : ⟨ G ⟩) → (x ·⟨ G ⟩ (y ·⟨ G ⟩ z)) ≡ ((x ·⟨ G ⟩ y) ·⟨ G ⟩ z)
+-- semigroup-assoc (_ , _ , _ , P) = P
 
--- Semigroup ·syntax
+-- -- Semigroup ·syntax
 
-module semigroup-·syntax (G : Semigroup {ℓ}) where
```

Comment:
@Saizan @guillaumebrunerie : Do you know if we need this kind of syntax modules when we use records? It seems to me like we don't as we can open the record and name the operator whatever we want?

(My Agda skills are quite rusty and because we've been using nested Sigmas for everything up until now I haven't really figured out how to work most efficiently with records yet)

Reply:
Indeed, if `Semigroup` is a record and `G : Semigroup`, then `Semigroup G` is a module that contains all the fields of `G`, so you don’t need to define such a module yourself.

Reply:
Great! Yet another reason to not use nested Sigma types